docs/language/functions.rst

```diff
@@ -353,6 +353,12 @@ values. Here is an example of an overloaded function:
 In the function foo, abs() is called with an ``int64`` so the second implementation
 of the function abs() is called.
 
+.. note::
+    Some target runtimes require unique function names (e.g. Substrate). In that case,
+    the names of overloaded functions will be mangled in the ABI as follows:
+    The function name will be concatenated with its argument type, sperated by underscores.
+    Tuple types are preceded with an additional underscore.
```

Comment:
Tuples don't exist in Solidity, only in the eth abi encoding. I would say structs.

Also I would provide a much better description of what the mangled names look like, without depending too much on how the signature is written. This will help users understand what their mangled name will be. Also include some examples and also mention how arrays, enums. structs etc are mangled.

Reply:
>  description of what the mangled names look like, without depending too much on how the signature 

Hmm, I tried exactly that by writing `function name will be concatenated with its argument type` :grin: I am not sure how I could make that more general (except that I wanted to write `with all of its argument types`)?

I can include some examples. I think together with the examples someone must be brain-dead to not get it.

Reply:
well enums are replaced with `uint8` (I think), structs with a list of members (with an extra underscore), and I don't know arrays are handled (fixed length or dynamic). None of that is mentioned